mobile/examples/whisper/azure/android/app/src/main/java/ai/onnxruntime/example/whisperAzure/AudioTensorSource.kt

```diff
@@ -0,0 +1,202 @@
+                val outputStream = FileOutputStream(outputFilePath)
+
+                // Write the WAV header
+                outputStream.write(header)
+
+                // Write the PCM audio data
+                outputStream.write(pcmData)
+
+                outputStream.close()
+            } catch (e: IOException) {
+                e.printStackTrace()
+            }
+        }
+
+        private fun createWavHeader(dataSize: Int, sampleRate: Int, bitsPerSample: Int, channels: Int): ByteArray {
+            val headerSize = 44
+            val totalSize = dataSize + headerSize - 8 // Subtract 8 for the RIFF chunk size
+
+            val header = ByteArray(headerSize)
+
+            // RIFF chunk descriptor
+            header[0] = 'R'.code.toByte()
+            header[1] = 'I'.code.toByte()
+            header[2] = 'F'.code.toByte()
+            header[3] = 'F'.code.toByte()
+            ByteBuffer.wrap(header, 4, 4).order(ByteOrder.LITTLE_ENDIAN).putInt(totalSize)
+
+            // Format chunk
+            header[8] = 'W'.code.toByte()
+            header[9] = 'A'.code.toByte()
+            header[10] = 'V'.code.toByte()
+            header[11] = 'E'.code.toByte()
+            header[12] = 'f'.code.toByte()
+            header[13] = 'm'.code.toByte()
+            header[14] = 't'.code.toByte()
+            header[15] = ' '.code.toByte()
+            ByteBuffer.wrap(header, 16, 4).order(ByteOrder.LITTLE_ENDIAN).putInt(16)
+            ByteBuffer.wrap(header, 20, 2).order(ByteOrder.LITTLE_ENDIAN).putShort(1.toShort()) // PCM format
+            ByteBuffer.wrap(header, 22, 2).order(ByteOrder.LITTLE_ENDIAN).putShort(channels.toShort())
+            ByteBuffer.wrap(header, 24, 4).order(ByteOrder.LITTLE_ENDIAN).putInt(sampleRate)
+            ByteBuffer.wrap(header, 28, 4).order(ByteOrder.LITTLE_ENDIAN)
+                .putInt(sampleRate * channels * bitsPerSample / 8) // Byte rate
+            ByteBuffer.wrap(header, 32, 2).order(ByteOrder.LITTLE_ENDIAN)
+                .putShort((channels * bitsPerSample / 8).toShort()) // Block align
+            ByteBuffer.wrap(header, 34, 2).order(ByteOrder.LITTLE_ENDIAN).putShort(bitsPerSample.toShort())
+
+            // Data chunk
+            header[36] = 'd'.code.toByte()
+            header[37] = 'a'.code.toByte()
+            header[38] = 't'.code.toByte()
+            header[39] = 'a'.code.toByte()
+            ByteBuffer.wrap(header, 40, 4).order(ByteOrder.LITTLE_ENDIAN).putInt(dataSize)
+
+            return header
+        }
+
+        fun fromRawWavBytes(rawBytes: ByteArray): OnnxTensor {
+            val rawByteBuffer = ByteBuffer.wrap(rawBytes)
+            // TODO handle big-endian native order...
+            if (ByteOrder.nativeOrder() != ByteOrder.LITTLE_ENDIAN) {
+                throw NotImplementedError("Reading Wav data is only supported when native byte order is little-endian.")
+            }
```

Comment:
do we need to enforce that the host byte order is little endian? we are only dealing with raw bytes and not wider types where byte order is significant.

Reply:
removed!